Description: Use the debian packaged version of rsyntaxtextarea
 Debian ships an old version of rsyntaxtextarea that is usable if the
 arduino code is slightly patched.
 .
 arduino (2:1.8.5+dfsg-1) UNRELEASED; urgency=medium
 .
   [ Geert Stappers ]
   * Non-maintainer upload.
   * New upstream version (Closes: #780706)
   * watch file revisited, s/google code/git hub/
   * README.source:  new file, to document some how our workflow
   * Documented '+dfsg' in README.source
   * raised compat version from 8 to 9
   * control:
     - build depend on libastyle-dev
     - build depends for .jar files that upstream has in their tarball
     - build depend on unzip
   * Various patches refreshed
   * New patches:
     - Libastyle is packaged
     - No Mac OS X  Think Different
     - Seperate reference
 .
   [ Ignacio Losiggio ]
   * New upstream version
   * Various patches refreshed
Author: Ignacio Losiggio <iglosiggio@gmail.com>
Bug-Debian: https://bugs.debian.org/780706
Last-Update: 2018-01-02

--- arduino-1.8.5+dfsg.orig/app/src/processing/app/EditorTab.java
+++ arduino-1.8.5+dfsg/app/src/processing/app/EditorTab.java
@@ -137,7 +137,7 @@ public class EditorTab extends JPanel im
     Gutter gutter = scrollPane.getGutter();
     gutter.setBookmarkingEnabled(false);
     //gutter.setBookmarkIcon(CompletionsRenderer.getIcon(CompletionType.TEMPLATE));
-    gutter.setIconRowHeaderInheritsGutterBackground(true);
+    //gutter.setIconRowHeaderInheritsGutterBackground(true);
 
     return scrollPane;
   }
@@ -269,7 +269,10 @@ public class EditorTab extends JPanel im
             referenceItem.setEnabled(referenceFile != null);
     
             int offset = textarea.getCaretPosition();
-            org.fife.ui.rsyntaxtextarea.Token token = RSyntaxUtilities.getTokenAtOffset(textarea, offset);
+            /* Use old RSyntaxVersion */
+            RSyntaxDocument doc = (RSyntaxDocument)textarea.getDocument();
+            org.fife.ui.rsyntaxtextarea.Token tokenlist = doc.getTokenListForLine(doc.getDefaultRootElement().getElementIndex(offset));
+            org.fife.ui.rsyntaxtextarea.Token token = RSyntaxUtilities.getTokenAtOffset(tokenlist, offset);
             if (token != null && token.isHyperlink()) {
               openURLItem.setEnabled(true);
               openURLItem.setActionCommand(token.getLexeme());
--- arduino-1.8.5+dfsg.orig/app/src/processing/app/syntax/SketchTextAreaEditorKit.java
+++ arduino-1.8.5+dfsg/app/src/processing/app/syntax/SketchTextAreaEditorKit.java
@@ -189,7 +189,7 @@ public class SketchTextAreaEditorKit ext
 
       org.fife.ui.rsyntaxtextarea.Token t = doc.getTokenListForLine(start);
       int languageIndex = t != null ? t.getLanguageIndex() : 0;
-      String[] startEnd = doc.getLineCommentStartAndEnd(languageIndex);
+      String[] startEnd = doc.getLineCommentStartAndEnd();
 
       if (startEnd == null) {
         UIManager.getLookAndFeel().provideErrorFeedback(textArea);
